pkg/pipelineascode/match.go

```diff
@@ -195,13 +201,32 @@ func (p *PacRun) getPipelineRunsFromRepo(ctx context.Context, repo *v1alpha1.Rep
 		return nil, nil
 	}
 
+	// finally resolve with fetching the remote tasks (if enabled)
+	if p.run.Info.Pac.RemoteTasks {
+		// only resolve on the matched pipelineruns
+		types.PipelineRuns = nil
+		for _, match := range matchedPRs {
+			for pr := range pipelineRuns {
+				if match.PipelineRun.Name == pipelineRuns[pr].Name {
+					types.PipelineRuns = append(types.PipelineRuns, pipelineRuns[pr])
+				}
+			}
+		}
+		pipelineRuns, err = resolve.Resolve(ctx, p.run, p.logger, p.vcx, types, p.event, &resolve.Opts{
+			GenerateName: true,
+			RemoteTasks:  true,
+		})
+		if err != nil {
+			p.eventEmitter.EmitMessage(repo, zap.ErrorLevel, "RepositoryFailedToMatch", fmt.Sprintf("failed to match pipelineRuns: %s", err.Error()))
+			return nil, err
+		}
+	}
+
 	err = changeSecret(pipelineRuns)
 	if err != nil {
 		return nil, err
 	}
-
-	// Match the PipelineRun with annotation
-	matchedPRs, err := matcher.MatchPipelinerunByAnnotation(ctx, p.logger, pipelineRuns, p.run, p.event, p.vcx)
+	matchedPRs, err = matcher.MatchPipelinerunByAnnotation(ctx, p.logger, pipelineRuns, p.run, p.event, p.vcx)
```

Comment:
Do we need this again ?

Reply:
yes because this time we are fetching the pipelineruns (for the only the one that have matched) with remote annotations and "resolve/embed" them inside the PipelineRun...

Reply:
as you are calling MatchPipelinerunByAnnotation function at the end once correct PipelineRuns are available  
Then why this require https://github.com/openshift-pipelines/pipelines-as-code/pull/1371/files#diff-c985856260500332bdba26c6f6c75a2e3c365e0ecd4272b92762e5de820dc2ccR188-R195 :thinking: 

Is that because to get matchedPRs here https://github.com/openshift-pipelines/pipelines-as-code/pull/1371/files#diff-c985856260500332bdba26c6f6c75a2e3c365e0ecd4272b92762e5de820dc2ccR208 ?

Reply:
we first match PR without getting remote tasks, making sure there is one that at least one matched the event
we do the matching/resolution again with the remote tasks included,

matchprbyannotation may be renamed matchprbyannotationandResolveThem maybe but that may be too verbose..